<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from A-2085-05-simple.xacro         | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="A-2085-05-simple" version="1.0">
  <!-- This is a single xacro file you can include to get
  all of the HEBI components -->
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="M1"/>
  </joint>
  <link name="M1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="red">
        <color rgba=".82 0.06 0.06 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.48"/>
      <origin rpy="0 0 0" xyz="-0.0145 -0.0031 0.0242"/>
      <inertia ixx="0.000246" ixy="0.0000444" ixz="0.0000266" iyy="0.000380" iyz="0.00000422" izz="0.000463"/>
    </inertial>
  </link>
  <joint name="M1_to_shoulder_bracket" type="continuous">
    <parent link="M1"/>
    <child link="shoulder_bracket"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
    <axis xyz="0 0 1"/>
    <limit effort="20" velocity="3.142"/>
    <dynamics damping="0.5" friction="0.5"/>
  </joint>
  <link name="shoulder_bracket">
    <visual>
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0 0 0.005"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2149-01.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 -.03 .04"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2150-01R.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0 0 0.005"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2149-01.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 -.03 .04"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2150-01R.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <!-- NOTE: inertia is just modelled as small sphere, and could be improved. -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.01875 .0275"/>
      <mass value="0.215"/>
      <inertia ixx="0.00030960000000000004" ixy="0.0" ixz="0.0" iyy="0.00030960000000000004" iyz="0.0" izz="0.00030960000000000004"/>
    </inertial>
  </link>

  <joint name="shoulder_bracket_to_M2" type="fixed">
    <parent link="shoulder_bracket"/>
    <child link="M2"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 -.0375 .055"/>
  </joint>
  <link name="M2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="red">
        <color rgba=".82 0.06 0.06 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.5"/>
      <origin rpy="0 0 0" xyz="-0.0145 -0.0031 0.0242"/>
      <inertia ixx="0.000246" ixy="0.0000444" ixz="0.0000266" iyy="0.000380" iyz="0.00000422" izz="0.000463"/>
    </inertial>
  </link>

  <joint name="M2_to_shoulder_elbow" type="continuous">
    <parent link="M2"/>
    <child link="shoulder_elbow"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
    <axis xyz="0 0 1"/>
    <limit effort="38" velocity="1.571"/>
    <dynamics damping="0.5" friction="0.5"/>
  </joint>
  <link name="shoulder_elbow">
    <!-- Input Cap -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2038-02.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2038-02.stl"/>
      </geometry>
    </collision>
    <!-- Link Pipe -->
    <visual>
      <origin rpy="0 1.57075 0" xyz="0.1625 0 0.02"/>
      <geometry>
        <cylinder length="0.29960000000000003" radius=".016"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 1.57075 0" xyz="0.1625 0 0.02"/>
      <geometry>
        <cylinder length="0.29960000000000003" radius=".016"/>
      </geometry>
    </collision>
    <!-- Output Cap -->
    <visual>
      <origin rpy="-3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2039-02.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2039-02.stl"/>
      </geometry>
    </collision>
    <!-- Approximate inertia as thin rod about x axis -->
    <inertial>
      <origin rpy="0 0 0" xyz="0.1625 0 0.02"/>
      <mass value="0.402"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0035384375000000003" iyz="0.0" izz="0.0035384375000000003"/>
    </inertial>
  </link>
  
  <joint name="shoulder_elbow_to_M3" type="fixed">
    <parent link="shoulder_elbow"/>
    <child link="M3"/>
    <origin rpy="3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
  </joint>
  <link name="M3">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="red">
        <color rgba=".82 0.06 0.06 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X8.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.48"/>
      <origin rpy="0 0 0" xyz="-0.0145 -0.0031 0.0242"/>
      <inertia ixx="0.000246" ixy="0.0000444" ixz="0.0000266" iyy="0.000380" iyz="0.00000422" izz="0.000463"/>
    </inertial>
  </link>
  <joint name="M3_to_elbow" type="continuous">
    <parent link="M3"/>
    <child link="elbow_wrist"/>
    <origin rpy="0 0 0" xyz="0 0 0.045"/>
    <axis xyz="0 0 1"/>
    <limit effort="20" velocity="3.142"/>
    <dynamics damping="0.5" friction="0.5"/>
  </joint>
  <link name="elbow_wrist">
    <!-- Input Cap -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2038-02.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2038-02.stl"/>
      </geometry>
    </collision>
    <!-- Link Pipe -->
    <visual>
      <origin rpy="0 1.57075 0" xyz="0.1625 0 0.02"/>
      <geometry>
        <cylinder length="0.29960000000000003" radius=".016"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 1.57075 0" xyz="0.1625 0 0.02"/>
      <geometry>
        <cylinder length="0.29960000000000003" radius=".016"/>
      </geometry>
    </collision>
    <!-- Output Cap -->
    <visual>
      <origin rpy="-3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2039-02.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/A-2039-02.stl"/>
      </geometry>
    </collision>
    <!-- Approximate inertia as thin rod about x axis -->
    <inertial>
      <origin rpy="0 0 0" xyz="0.1625 0 0.02"/>
      <mass value="0.402"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0035384375000000003" iyz="0.0" izz="0.0035384375000000003"/>
    </inertial>
  </link>
  
  <!--joint name="elbow_wrist1/OUTPUT_JOINT" type="fixed">
    <parent link="elbow_wrist1/INPUT_INTERFACE"/>
    <child link="Arm/J4_wrist1/INPUT_INTERFACE"/>
    <origin rpy="3.141592653589793 0 0" xyz="0.325 -2.4492935982947064e-18 0.0"/>
  </joint>
  <link name="Arm/J4_wrist1/INPUT_INTERFACE">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X5.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="red">
        <color rgba=".82 0.06 0.06 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/X5.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.315"/>
      <origin rpy="0 0 0" xyz="-0.0142 -0.0031 0.0165"/>
      <inertia ixx="0.00015" ixy="0.0000341" ixz="0.0000118" iyy="0.000255" iyz="0.00000229" izz="0.000350"/>
    </inertial>
  </link>
  <gazebo reference="Arm/J4_wrist1/INPUT_INTERFACE">
    <material>Gazebo/Red</material>
  </gazebo>
  <joint name="Arm/J4_wrist1" type="continuous">
    <parent link="Arm/J4_wrist1/INPUT_INTERFACE"/>
    <child link="wrist2_bracket/INPUT_INTERFACE"/>
    <origin rpy="0 0 0" xyz="0 0 0.031"/>
    <axis xyz="0 0 1"/>
    <limit effort="2.5" velocity="9.425"/>
    <dynamics damping="0.1" friction="0.1"/>
  </joint>
  <link name="wrist2_bracket/INPUT_INTERFACE">
    <visual>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2132-01R.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="beige">
        <color rgba="0.96 0.945 0.871 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://hebi_description/meshes/PM-2132-01R.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 -.0215 .02"/>
      <mass value="0.1"/>
      <inertia ixx="0.000144" ixy="0.0" ixz="0.0" iyy="0.000144" iyz="0.0" izz="0.000144"/>
    </inertial>
  </link>
  <gazebo reference="wrist2_bracket/INPUT_INTERFACE">
    <visual>
      <material>
        <ambient>0.96 0.945 0.871 1</ambient>
        <diffuse>0.96 0.945 0.871 1</diffuse>
      </material>
    </visual>
  </gazebo>
  <joint name="wrist2_bracket/OUTPUT_INTERFACE" type="fixed">
    <parent link="wrist2_bracket/INPUT_INTERFACE"/>
    <child link="Arm/J5_wrist2/INPUT_INTERFACE"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 -.043 .04"/>
  </joint-->
</robot>

